numpy/core/tests/test_multiarray.py

```diff
@@ -8262,6 +8262,9 @@ class MyAlwaysEqualNew(MyAlwaysEqual):
         assert_equal(array != my_always_equal, 'ne')
 
 
+@pytest.mark.xfail(sys.platform == 'cygwin',
+                   reason='abs(inf+nanj) short-circuits to inf',
+                   strict=True)
```

Comment:
Is this just the `clongdouble` loop iteration below failing?

Reply:
It would be great to `parameterize` this test to make the xfail more precise.

Reply:
```python
>>> np.abs(np.array([complex(np.inf, np.nan)], np.complex64))
array([inf], dtype=float32)
>>> np.abs(np.array([complex(np.inf, np.nan)], np.complex128))
array([inf])
>>> np.abs(np.array([complex(np.inf, np.nan)], np.complex256))
array([nan], dtype=float128)
```
As near as I can tell, `clongdouble` is the only one getting this right.

It looks relatively easy to parametrize.  I'll look into that.